index.js

```diff
@@ -273,61 +281,96 @@ async function handleSensorRequest(currentLocationId, radarType) {
     await redis.addStateMachineData(currentState, currentLocationId)
 
     if (VOIDSTATES.includes(currentState)) {
-      const latestSession = await db.getMostRecentSession(currentLocationId)
-
-      if (typeof latestSession !== 'undefined') {
-        // Checks if no session exists for this location yet.
-        if (latestSession.end_time == null) {
-          // Checks if session is open.
-          await db.updateSessionState(latestSession.sessionid, currentState)
+      let client
+      try {
+        client = await db.beginTransaction()
+        const latestSession = await db.getMostRecentSession(currentLocationId, client)
+
+        if (typeof latestSession !== 'undefined') {
+          // Checks if no session exists for this location yet.
+          if (latestSession.endTime == null) {
+            // Checks if session is open.
+            await db.updateSessionState(latestSession.sessionid, currentState, client)
+          }
+        }
+        await db.commitTransaction(client)
```

Comment:
This didn't have a transaction before, but I think it makes sense for it to have one.

Reply:
agree